web-share/share-empty.https.html

```diff
@@ -0,0 +1,34 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <title>WebShare Test: Share no known fields</title>
+    <script src="/resources/testharness.js"></script>
+    <script src="/resources/testharnessreport.js"></script>
+    <script src="resources/manual-helper.js"></script>
+  </head>
+  <body>
+    <script>
+        promise_test(t => {
+          return promise_rejects(t, new TypeError(), navigator.share());
+        }, 'share with no arguments (same as empty dictionary)');
+
+        promise_test(t => {
+          return promise_rejects(t, new TypeError(), navigator.share({}));
```

Comment:
I think making the empty dictionary be a successful (but empty) share was a mistake. Especially if you consider future expansion: if we add a new field and someone in the future *exclusively* uses that field, then how should it behave on old browsers that don't know about that field? We decided it's best for them to throw a TypeError rather than succeeding silently (and will provide feature-detection tools so devs can detect that case before calling `share`).

Full rationale is [here](https://wicg.github.io/web-share/#extensibility-of-this-api).